### PR TITLE
refactor: auth flow to ant design

### PR DESCRIPTION
**Pull request by tristanhcole**
**Describe the Pull Request**
Refactor auth flow to ant design. Refactors the login form, register form, request password email form and reset password form. Swaps from `react-password-strength` to `antd-password-input-strength` that use the same password strength validator.

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoBlockchain/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by tristanhcole** [comment]
[refactor: auth flow](https://trello.com/c/WiGfcqQd/872-refactor-auth-flow)

File: app/server/api/auth_api.py [hunk 1]
```diff
@@ -627,7 +627,8 @@ def post(self):
 
         # Check authorisation using regular auth
         elif auth_header and auth_header != 'null' and old_password:
-            auth_token = auth_header.split(" ")[0]
+            split_header = auth_header.split("|")
+            auth_token = split_header[0]
```
Comment (tristanhcole) [inline app/server/api/auth_api.py:631]: This was handling auth tokens incorrectly, updated to support TFA

File: app/client/components/auth/loginForm.jsx [hunk 2]
```diff
@@ -25,58 +22,22 @@ const mapDispatchToProps = dispatch => {
...
-    });
-    if (e.nativeEvent.keyCode != 13) return;
-    this.attemptlogin();
-  }
-
-  onClick() {
-    this.attemptlogin();
-  }
+  onFinishFailed = errorInfo => {
+    console.log("Failed:", errorInfo);
```
Comment (michielderoos) [inline app/client/components/auth/loginForm.jsx:39]: Should we be logging here?

File: app/client/components/auth/registerForm.jsx [hunk 3]
```diff
@@ -23,224 +21,139 @@ const mapDispatchToProps = dispatch => {
...
+            }
+          })
+        ]}
+        aria-label="password"
+        dependencies={["retypePassword"]}
+      >
+        <PasswordInput
+          inputProps={{
+            placeholder: "Password",
+            minLength: "8",
```
Comment (michielderoos) [inline app/client/components/auth/registerForm.jsx:120]: Not sure why, but it's rejecting an 8 character password (error message goes away on character 9)
image
image

**Review by michielderoos — CHANGES_REQUESTED** [review]
One little fix, but LGTM otherwise!

File: app/client/components/auth/loginForm.jsx (hunk 2 above)
Comment (tristanhcole) [reply app/client/components/auth/loginForm.jsx:39]: It seems fine to me

File: app/client/components/auth/registerForm.jsx [hunk 4]
```diff
@@ -23,224 +21,138 @@ const mapDispatchToProps = dispatch => {
...
+        rules={[
+          ({ getFieldValue }) => ({
+            validator(_, value) {
+              if (!value) {
+                return Promise.reject("Please enter your password!");
+              }
+              if (value && value.length < 8) {
+                return Promise.reject(
+                  "Password must be at least 8 characters long"
+                );
```
Comment (tristanhcole) [inline app/client/components/auth/registerForm.jsx:105]: Here

File: app/client/components/auth/registerForm.jsx (hunk 3 above)
Comment (tristanhcole) [reply app/client/components/auth/registerForm.jsx]: This was the issue: https://github.com/teamsempo/SempoBlockchain/pull/739#discussion_r684902854

**Review by michielderoos — APPROVED** [review]
👍

**Pull request merged**